### PR TITLE
Suppress compile time warnings

### PR DESCRIPTION
**Pull request by ashie**
I've added `-Wall` to `CFLAGS` and suppress compile time warnings found by it.
In addition, remove some unused functions.

Some of such warnings are harmless, but some of them cause actual issues.
Please check patches one by one.

### CODE REVIEW

File: ext/winevt/winevt_channel.c [hunk 1]
```diff
@@ -63,7 +63,7 @@ rb_winevt_channel_each(VALUE self)
     winevtChannel->channels = hChannels;
   } else {
     _snprintf_s(
-      errBuf, 256, _TRUNCATE, "Failed to enumerate channels with %s\n", GetLastError());
```
Comment (cosmo0920) [inline ext/winevt/winevt_channel.c]: We should use `sizeof(errBuf)` instead of 256?

**Comment by cosmo0920** [comment]
I've confirmed that these changes does not break existing feature.

File: ext/winevt/winevt_channel.c (hunk 1 above)
Comment (ashie) [reply ext/winevt/winevt_channel.c]: Ah, yes.

File: test/test_winevt.rb [hunk 2]
```diff
@@ -47,6 +47,13 @@ def test_seek
       assert_true(@query.seek(:last))
       assert_true(@query.seek("last"))
     end
+
+    def test_seek_with_invalid_flag
+      error = assert_raises ArgumentError do
+        @query.seek("hoge");
```
Comment (cosmo0920) [inline test/test_winevt.rb]: hoge.... :sweat_smile:

File: ext/winevt/winevt_channel.c (hunk 1 above)
Comment (ashie) [reply ext/winevt/winevt_channel.c]: We'll use `_countof()` which can be used in most cases both ANSI and UNICODE API.
I'll do it in another PR.

Comment (ashie) [reply ext/winevt/winevt_channel.c]: I committed it directly to the master branch.

**Pull request merged**